libp2p/pubsub/pubsub.py

```diff
@@ -361,4 +436,4 @@ def _mark_msg_seen(self, msg: rpc_pb2.Message) -> None:
     def _is_subscribed_to_msg(self, msg: rpc_pb2.Message) -> bool:
         if not self.my_topics:
             return False
-        return all([topic in self.my_topics for topic in msg.topicIDs])
+        return any([topic in self.my_topics for topic in msg.topicIDs])
```

Comment:
If you change the inner comprehension to use `()` then it will be a generator allowing `any` to return early on the first match.